M3-3865 Adding aria support to all entity Table components

WilkinsKa1 opened the pull request:
## Description

Tables weren't as accessible as they could be for users using the built-in MacOS Voiceover. I have added the `aria-sort` attribute to the sortable table header cell component and have also added the attributes `aria-colindex` and `aria-rowindex` to all major entity landing pages that had tables present (Linodes, NodeBalancers, Volumes, OBJ, Domains, LKE, Images, StackScripts)

Todo:
- [x] Volumes, StackScripts, Images all declare extra +1 column, +1 row (regardless of GBT enabling)

## Type of Change
- Non breaking change ('update')

## Note to Reviewers

Please test using Voiceover with Safari in addition to NVDA (if possible) to ensure there aren't any regressions. Not all assistive technologies interpret ARIA the same way for any given element (Fun!). Be sure to test group by tag toggle as well for entities with that feature present to ensure there aren't any functional regressions. Also please test the hover state for tables where it is present, I had to modify the way we were handling that as Voiceover was misinterpreting the pseudo element as a column.

acourdavault commented on packages/manager/src/components/TableRow/TableRow.tsx:
```diff
@@ -24,30 +24,20 @@ type ClassNames =
+      //   borderLeft: `1px solid transparent`,
+      //   paddingLeft: 4
+      // },
       '&:hover': {
```
@WilkinsKa1 i guess we should delete these commented line, or if we want to let them for future review, maybe add a comment for context at the top about why this is commented.
If you don t have time, let s create a jira task

WilkinsKa1 replied:
These should be good now, I think the file wasn't updating on git

Jskobos commented on packages/manager/src/features/linodes/LinodesLanding/TableWrapper.tsx:
```diff
@@ -21,16 +21,26 @@ const styles = (theme: Theme) =>
 
-type CombinedProps = Omit<OrderByProps, 'data'> & WithStyles<ClassNames>;
+interface Props {
+  data: any[];
```
Is this always a list of Linodes? if so, could be typed as Linode[]

Jskobos replied:
Ignore this ^^^ I changed my mind after finding a few of these, it's probably better to pass the length directly as a number rather than all of the data.

Jskobos commented on packages/manager/src/features/Volumes/VolumeTableWrapper.tsx:
```diff
@@ -23,6 +23,7 @@ const styles = (theme: Theme) =>
 
 interface VolumeTableWrapperProps {
   isVolumesLanding: boolean;
+  data: any[];
```
Same as above, should probably be `Volume[]`

Jskobos replied:
or now that I think of it, you should probably just pass down the length directly here, so it's `dataLength: number`

Jskobos commented on packages/manager/src/features/NodeBalancers/NodeBalancersLanding/NodeBalancersTableWrapper.tsx:
```diff
@@ -21,18 +21,27 @@ const styles = (theme: Theme) =>
 
-type CombinedProps = Omit<OrderByProps, 'data'> & WithStyles<ClassNames>;
+interface Props {
+  data: any[];
```
Same as elsewhere, pass the length down directly rather than all of the data

Jskobos commented on packages/manager/src/features/Domains/DomainsTableWrapper.tsx:
```diff
@@ -21,16 +21,26 @@ const styles = (theme: Theme) =>
 
-type CombinedProps = Omit<OrderByProps, 'data'> & WithStyles<ClassNames>;
+interface Props {
+  data: any[];
```
As elsewhere

Jskobos commented on packages/manager/src/components/TableRow/TableRow.tsx:
```diff
@@ -61,11 +51,12 @@ const styles = (theme: Theme) =>
-        borderColor: theme.palette.primary.light
-      },
+      boxShadow: `inset 3px 0 0 ${theme.bg.lightBlue}`,
+      // '&:before': {
```
Probably just delete these after everything's finalized

WilkinsKa1 replied:
This should be good now, I think the file wasn't updating for some reason on the PR.

WilkinsKa1 commented:
@Jskobos I addressed your comments, please take a look when you get a chance again!

@acourdavault I made some adjustments to OBJ, if you could please retry with voiceover + Safari. Sometimes using any other browser with Voiceover will deliver inconsistent results. 

Thanks for reviewing!
